### PR TITLE
DDF-1865 Fixes for Fortify Critical Issues

### PR DESCRIPTION
**Pull request by jckilmer**
(cherry picked from commit 937d5cf539a1ad383f348aff657966f2739e0716)

@pklinef @kcwire @shaundmorris @stustison @vinamartin @gordocanchola @djblue @clockard @brianfelix @mcalcote @oscaritoro

These are the same changes that have already been approved and merged to codice/master.

### CODE REVIEW

**Comment by oscaritoro** [comment]
LGTM... Just need to confirm with fortify that there are no other critical issues on this branch. :+1:

**Comment by clockard** [comment]
LGTM

**Comment by vinamartin** [comment]
LGTM

**Comment by brianfelix** [comment]
LGTM

**Comment by ahoffer** [comment]
+1

File: catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/java/org/codice/ddf/spatial/geocoder/endpoint/JSONPValidator.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+package org.codice.ddf.spatial.geocoder.endpoint;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class JSONPValidator {
```
Comment (pklinef) [inline catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/java/org/codice/ddf/spatial/geocoder/endpoint/JSONPValidator.java]: Acronyms like JSONP should be camel cased as if they are words so it should be `JsonpValidator`.

File: platform/security/cas/security-cas-client/src/main/java/org/codice/ddf/security/handler/cas/TestPage.java [hunk 2]
```diff
@@ -1,187 +0,0 @@
...
-import org.slf4j.LoggerFactory;
-
-/**
- * Proxy granting ticket callback used by the CAS server. Nothing really to do here, we just need
- * something running at the callback url so the CAS server doesn't get a HTTP 404 error. This is
- * also used as a test page.
- *
- * @see javax.servlet.http.HttpServlet
- */
-public class TestPage extends HttpServlet {
```
Comment (pklinef) [inline platform/security/cas/security-cas-client/src/main/java/org/codice/ddf/security/handler/cas/TestPage.java]: Why is this no longer needed?

**Comment by roelens8** [comment]
:+1:

**Comment by jckilmer** [comment]
I pushed up a few additional changes because there are a few more Fortify critical issues on this branch that were not present on master. I'm currently running a Fortify scan on the changes to verify the issues have been resolved.

File: platform/security/cas/security-cas-client/src/main/java/org/codice/ddf/security/handler/cas/TestPage.java (hunk 2 above)
Comment (oscaritoro) [inline platform/security/cas/security-cas-client/src/main/java/org/codice/ddf/security/handler/cas/TestPage.java]: I believe both Shaun and Scott stated this was no longer used and could be safely removed.

File: catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/java/org/codice/ddf/spatial/geocoder/endpoint/JSONPValidator.java (hunk 1 above)
Comment (oscaritoro) [inline catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/java/org/codice/ddf/spatial/geocoder/endpoint/JSONPValidator.java]: The one that got checked into master should probably be renamed as well then.

Comment (oscaritoro) [inline catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/java/org/codice/ddf/spatial/geocoder/endpoint/JSONPValidator.java]: Should the method name also get changed?

File: platform/security/sts/security-sts-ldapclaimshandler/src/main/java/ddf/security/sts/claimsHandler/ContextSourceDecrypt.java [hunk 3]
```diff
@@ -47,8 +47,7 @@ public void setEncryptionService(EncryptionService encryptionService) {
     public void setPassword(String password) {
         if (encryptionService == null) {
             LOGGER.error(
-                    "The ContextSourceDecrypt has a null Encryption Service.  Unable to attempt to decrypt the encrypted password: {}.  Setting decrypted password to null.",
-                    password);
+                    "The ContextSourceDecrypt has a null Encryption Service.  Unable to attempt to decrypt the encrypted password: [HIDDEN].  Setting decrypted password to null.");
             this.password = password;
```
Comment (oscaritoro) [inline platform/security/sts/security-sts-ldapclaimshandler/src/main/java/ddf/security/sts/claimsHandler/ContextSourceDecrypt.java]: Not sure that this is a problem, but this is not really doing what the error message claims (it's setting the password to whatever got passed in without decrypting, not setting it to null.)

**Comment by mcalcote** [comment]
Agree with @oscaritoro that the error message in ContextSourceDecrypt.java should reflect actual functionality. Other than that, LGTM :+1:

**Comment by oscaritoro** [comment]
Updated master (see PR #612) based on the comments that @pklinef and @mcalcote did and then cherry picked the changes into this PR.

**Pull request merged**